index.src.html

```diff
@@ -2394,14 +2420,692 @@
+        2. Assign |selectedTrackIndex| to {{ImageDecoder/tracks}}
+            {{ImageTrackList/[[selected index]]}}.
+        3. Fire a simple event named {{ImageTrackList/change}} at the
+            {{ImageDecoder/tracks}} object.
+        4. For each |promise| in
+            {{ImageDecoder/[[pending metadata promises]]}}, resolve |promise|
+            and remove it from the list.
+
+: <dfn for=ImageDecoder>Update Tracks</dfn>
+:: A <dfn>track update struct</dfn> is a [=struct=] that consists of a
+    <dfn for="track update struct">track index</dfn> ({{unsigned long}})
+    and a <dfn for="track update struct">frame count</dfn>
+    ({{unsigned long}}).
+
+    Run these steps:
+    1. Assert {{ImageDecoder/[[tracks established]]}} is `true`.
+    2. Let |trackChanges| be a new [=list=].
+    3. Let |trackList| be a copy of {{ImageDecoder/tracks}}'
+        {{ImageTrackList/[[track list]]}}.
+    4. For each |track| in |trackList|:
+        1. Let |trackIndex| be  the position of |track| in |trackList|.
+        2. Let |latestFrameCount| be the frame count as indicated by
+            {{ImageDecoder/[[encoded data]]}} for the track corresponding to
+            |track|.
+        3. Assert that |latestFrameCount| is greater than or equal to
+            `track.frameCount`.
+        4. If |latestFrameCount| is greater than `track.frameCount`:
+            1. Let |change| be a [=track update struct=] whose
+                [=track update struct/track index=] is |trackIndex| and
+                [=track update struct/frame count=] is |latestFrameCount|.
+            2. Append |change| to |tracksChanges|.
+    5. If |tracksChanges| is [=list/empty=], abort these steps.
+    6. For each <var ignore=''>update</var> in |trackChanges|:
+        1. Let |updateTrack| be the {{ImageTrack}} at position
+            `update.trackIndex` within {{ImageDecoder/tracks}}'
+            {{ImageTrackList/[[track list]]}}.
+        2. Assign `update.frameCount` to |updateTrack|'s
+            {{ImageTrack/[[frame count]]}}.
+    7. Fire a simple event named {{ImageTrackList/change}} at the
+        {{ImageDecoder/tracks}} object.
+
+: <dfn for=ImageDecoder>Decode Complete Frame</dfn> (with |frameIndex| and
+    |promise|)
+:: 1. Assert that {{ImageDecoder/[[tracks established]]}} is `true`.
+    2. Assert that {{ImageDecoder/[[internal selected track index]]}} is not
+        `-1`.
+    3. Let |encodedFrame| be the encoded frame identified by |frameIndex| and
+        {{ImageDecoder/[[internal selected track index]]}}.
+    4. Wait for any of the following conditions to be true (whichever happens
+        first):
+        1. {{ImageDecoder/[[encoded data]]}} contains enough bytes to
+            completely decode |encodedFrame|.
+        2. {{ImageDecoder/complete}} is `true`.
+        3. {{ImageDecoder/[[fatal error]]}} is `true`.
+    5. If {{ImageDecoder/[[encoded data]]}} does not contain enough bytes to
```

Comment:
Probably meant to say if |complete| before this section? Maybe the text needs a FALLTHROUGH macro :p

Reply:
Sounds like yes you did mean to use the spec equivalent of FALLTHROUGH :)